### PR TITLE
2.x: fix concatMapEager should accept 0 for prefetch

### PR DESCRIPTION
**Pull request by nmorioka**
Fixes a bug that  Flowable.concatMapEager(mapper , maxConcurrency , prefetch) and Observable.concatMapEager(mapper , maxConcurrency , prefetch) operators will not accept 0 and negative numbers.

Reported in #5185.

### CODE REVIEW

File: src/main/java/io/reactivex/Flowable.java [hunk 1]
```diff
@@ -6780,7 +6780,6 @@ public final void blockingSubscribe(Subscriber<? super T> subscriber) {
     public final <R> Flowable<R> concatMapEager(Function<? super T, ? extends Publisher<? extends R>> mapper,
             int maxConcurrency, int prefetch) {
         ObjectHelper.verifyPositive(maxConcurrency, "maxConcurrency");
-        ObjectHelper.verifyPositive(prefetch, "prefetch");
```
Comment (akarnokd) [inline src/main/java/io/reactivex/Flowable.java]: No, the validations were right, you shouldn't call with non-positive value.

File: src/main/java/io/reactivex/Observable.java [hunk 2]
```diff
@@ -5975,7 +5975,6 @@ public final void blockingSubscribe(Observer<? super T> subscriber) {
             int maxConcurrency, int prefetch) {
         ObjectHelper.requireNonNull(mapper, "mapper is null");
         ObjectHelper.verifyPositive(maxConcurrency, "maxConcurrency");
-        ObjectHelper.verifyPositive(prefetch, "prefetch");
```
Comment (akarnokd) [inline src/main/java/io/reactivex/Observable.java]: No, the validations were right, you shouldn't call with non-positive value.

**Comment by akarnokd** [comment]
I believe a 0 breaks the internals of the operator and -1 has the completely opposite effect, taking more memory.

File: src/main/java/io/reactivex/Flowable.java [hunk 3]
```diff
@@ -6770,7 +6770,7 @@ public final void blockingSubscribe(Subscriber<? super T> subscriber) {
      * @param mapper the function that maps a sequence of values into a sequence of Publishers that will be
      *               eagerly concatenated
      * @param maxConcurrency the maximum number of concurrent subscribed Publishers
-     * @param prefetch hints about the number of expected source sequence values
+     * @param prefetch hints about the number of expected values from each inner Publisher
```
Comment (akarnokd) [inline src/main/java/io/reactivex/Flowable.java]: I think this should be reinforced:

```
hints about the number of expected values from each inner Publisher, must be positive
```

**Review by akarnokd — CHANGES_REQUESTED** [review]
The verifyPositive should be restored.
Optionally, you could add the ", positive" to the javadoc changes.

**Comment by nmorioka** [comment]
I am sorry with the wrong fix.
Is it only necessary to modify the document?

**Comment by akarnokd** [comment]
> Is it only necessary to modify the document?

Yes. Please modify only the javadoc.

**Comment by nmorioka** [comment]
Thank you! 

Also, I think that the test case name is wrong, so I modifed it.

**Review by akarnokd — APPROVED** [review]

**Review by vanniktech — APPROVED** [review]

**Comment by akarnokd** [comment]
Thanks for the contribution!

**Pull request merged**